File: examples/plugins/sessao-watcher/src/hooks/on_status.py
from claude_workspaces.plugin_api import HookContext, SessionStatusChangedPayload


async def handler(
    ctx: HookContext, payload: SessionStatusChangedPayload
) -> None:
    if payload.new_status != "awaiting-input":
        return

    threshold_min = await ctx.config.get("threshold_minutes")
    if payload.duration_ms < threshold_min * 60 * 1000:
        return

    session = await ctx.sessions.get(payload.session_id)
    await ctx.ui.notify(
        title="Sessão aguardando há muito tempo",
        body=f"{session.workspace_name}: {session.last_message or '(sem título)'}",
    )

File: examples/plugins/sessao-watcher/src/hooks/on-status.ts
import { HookContext, SessionStatusChangedPayload } from "@claude-workspaces/api";

export default async function (
  ctx: HookContext,
  payload: SessionStatusChangedPayload
): Promise<void> {
  if (payload.newStatus !== "awaiting-input") return;

  const thresholdMin = await ctx.config.get<number>("threshold_minutes");
  if (payload.durationMs < thresholdMin * 60 * 1000) return;

  const session = await ctx.sessions.get(payload.sessionId);
  await ctx.ui.notify({
    title: "Sessão aguardando há muito tempo",
    body: `${session.workspaceName}: ${session.lastMessage ?? "(sem título)"}`,
    actions: [{ id: "focus", label: "Abrir" }],
  });
}
